include/plugin/runtime-support.h

```diff
@@ -710,6 +709,95 @@ inline ::zeek::ValPtr to_val(const hilti::rt::Set<T>& s, ::zeek::TypePtr target)
     return zv;
 }
 
+
+namespace {
+template<typename, template<typename...> typename>
+struct is_instance_impl : public std::false_type {};
+
+template<template<typename...> typename U, typename... Ts>
+struct is_instance_impl<U<Ts...>, U> : public std::true_type {};
+} // namespace
+
+template<typename T, template<typename...> typename U>
+using is_instance = is_instance_impl<std::remove_cv_t<T>, U>;
+
+
+template<typename T>
+inline void set_record_field(::zeek::RecordVal* rval, ::zeek::IntrusivePtr<::zeek::RecordType>& rtype, int idx,
+                             const T& x) {
+    using NoConversionNeeded =
+        std::integral_constant<bool, std::is_same<T, int8_t>::value || std::is_same<T, int16_t>::value ||
+                                         std::is_same<T, int32_t>::value || std::is_same<T, int64_t>::value ||
+                                         std::is_same<T, uint8_t>::value || std::is_same<T, uint16_t>::value ||
+                                         std::is_same<T, uint32_t>::value || std::is_same<T, uint64_t>::value ||
+                                         std::is_same<T, double>::value || std::is_same<T, std::string>::value ||
+                                         std::is_same<T, bool>::value>;
+
+    using IsSignedInteger =
+        std::integral_constant<bool, std::is_same<T, hilti::rt::integer::safe<int8_t>>::value ||
+                                         std::is_same<T, hilti::rt::integer::safe<int16_t>>::value ||
+                                         std::is_same<T, hilti::rt::integer::safe<int32_t>>::value ||
+                                         std::is_same<T, hilti::rt::integer::safe<int64_t>>::value>;
+
+    using IsUnsignedInteger =
+        std::integral_constant<bool, std::is_same<T, hilti::rt::integer::safe<uint8_t>>::value ||
+                                         std::is_same<T, hilti::rt::integer::safe<uint16_t>>::value ||
+                                         std::is_same<T, hilti::rt::integer::safe<uint32_t>>::value ||
+                                         std::is_same<T, hilti::rt::integer::safe<uint64_t>>::value>;
+
+    if constexpr ( NoConversionNeeded::value )
+        rval->Assign(idx, x);
+    else if constexpr ( IsSignedInteger::value )
+#if ZEEK_VERSION_NUMBER >= 50200
+        rval->Assign(idx, static_cast<int64_t>(x.Ref()));
+#else
+        rval->Assign(idx, static_cast<int>(x.Ref()));
+#endif
+    else if constexpr ( IsUnsignedInteger::value )
+        rval->Assign(idx, static_cast<uint64_t>(x.Ref()));
+    else if constexpr ( std::is_same<T, hilti::rt::Bytes>::value )
+        rval->Assign(idx, x.str());
+    else if constexpr ( std::is_same<T, hilti::rt::Bytes>::value )
+        rval->Assign(idx, x.str());
```

Comment:
This duplicates the branch above. Should this have been something else or is this just left-over copy-n-paste?

Reply:
Left-over indeed.